sdk/keyvault/azure-keyvault-keys/tests/_test_case.py

```diff
@@ -5,15 +5,33 @@
 import os
 
 from azure.keyvault.keys._shared import HttpChallengeCache
+from azure.keyvault.keys._shared.client_base import ApiVersion, DEFAULT_VERSION
 from devtools_testutils import AzureTestCase
 from parameterized import parameterized
 import pytest
 from six.moves.urllib_parse import urlparse
 
 
+def _get_test_parameters():
+    combinations = []
+    hsm_supported_versions = {ApiVersion.V7_2_preview}
+    for api_version in ApiVersion:
+        if api_version in hsm_supported_versions:
+            combinations.append([api_version, True])
+        combinations.append([api_version, False])
+    return combinations
+
+
 def suffixed_test_name(testcase_func, param_num, param):
+    api_version = param.kwargs.get("api_version")
     suffix = "mhsm" if param.kwargs.get("is_hsm") else "vault"
-    return "{}_{}".format(testcase_func.__name__, parameterized.to_safe_name(suffix))
+    return "{}_{}_{}".format(
+        testcase_func.__name__, parameterized.to_safe_name(api_version), parameterized.to_safe_name(suffix)
+    )
+
+
+# parameters for test case parameterization, where [x, y] = [api_version, is_hsm]
+PARAMETER_COMBINATIONS = _get_test_parameters()
```

Comment:
Looks like you use this only to generate many copies of the same list. Could you do something like this instead?
```py
PARAMS = [param(api_version=p[0], is_hsm=p[1])) for p in _get_test_parameters()]

# then import PARAMS in test modules
@parameterized.expand(PARAMS, name_func=suffixed_test_name)
def test_...
```
Or like this, better yet and no need for `PARAMS`:
```py
test_all_versions = functools.partial(parameterized.expand, PARAMS, name_func=suffixed_test_name)

@test_all_versions()
def test_...
```

Reply:
Great ideas! I'll try them out and push an update